catalog/models.py

```diff
@@ -250,53 +250,27 @@ def exclude_by_products(self, products: Iterable[AbstractProduct]):
             .distinct()
         )
 
-    # @todo #SE747:30m  Rm `get_group_tags_pairs`.
-    #  Use `get_grouped_tags -> Dict[Group, List[Tag]]` instead.
-    #  Tags pairs already lead us to errors with groups order.
-    def get_group_tags_pairs(self) -> List[Tuple[TagGroup, List['Tag']]]:
+    def get_group_tags_pairs(self) -> Dict[TagGroup, List['Tag']]:
         """
         Return set of group_tag pairs with specific properties.
 
         Every pair contains tag group and sorted tag list.
-        It's sorted alphabetically or numerically.
-        Sort method depends of tag value type.
+        It's sorted alphabetically and then numerically.
         """
         # @todo #STB374:120m Move tag's value to separated field.
         #  Now we have fields like `tag.name == '10 м'`.
         #  But should have smth like this:
         #  `tag.value, tag.group.measure, tag.label == 10, 'м', '10 м'`.
         #  Right now we should do dirty hacks for tags comparing mech.
-        def int_or_str(value: str):
-            try:
-                return int(value.split(' ')[0])
-            except ValueError:
-                return value
-
-        def has_only_int_keys(tags: list):
-            for t in tags:
-                key = int_or_str(t.name)
-                if isinstance(key, str):
-                    return False
-            return True
-
-        grouped_tags = groupby(
-            self
-            .prefetch_related('group')
-            .order_by('group__position', 'group__name'),
-            key=attrgetter('group')
-        )
-        result = []
-        for group, tags_ in grouped_tags:
-            tags_ = list(tags_)
-            key = (
-                (lambda t: int_or_str(t.name))
-                if has_only_int_keys(tags_)
-                else attrgetter('name')
-            )
-            tags_ = sorted(tags_, key=key)
-            result.append((group, tags_))
-
-        return result
+        ordered = self.prefetch_related('group').order_by_alphanumeric()
+
+        grouped = OrderedDict()
```

Comment:
dict is already ordered in py3.6. py3.7 officially documented this fact.
defaultdict(list) might be more good for code below:

```
grouped = defaultdict(list)
for tag in ordered:
    grouped[tag.group].append(tag)
```

Reply:
>dict is already ordered in py3.6. py3.7 officially documented this fact.

it is an internal implementation detail, that we shouldn't relay upon. it may change in the future